src/triage/component/postmodeling/contrast/model_evaluator.py

```diff
@@ -0,0 +1,1103 @@
+                           m.hyperparameters,
+                           m.model_hash,
+                           m.train_end_time,
+                           m.train_matrix_uuid,
+                           m.training_label_timespan,
+                           mg.model_type,
+                           mg.model_config
+                        FROM model_metadata.models m
+                        JOIN model_metadata.model_groups mg
+                        USING (model_group_id)
+                        WHERE model_group_id = {self.model_group_id}
+                        AND model_id = {self.model_id}
+                    ),
+                    individual_model_id_matrices AS(
+                    SELECT DISTINCT ON (matrix_uuid)
+                           model_id,
+                           matrix_uuid,
+                           as_of_date
+                        FROM test_results.predictions
+                        WHERE model_id = ANY(
+                            SELECT model_id
+                            FROM individual_model_ids_metadata
+                        )
+                    )
+                    SELECT metadata.*, test.*
+                    FROM individual_model_ids_metadata AS metadata
+                    LEFT JOIN individual_model_id_matrices AS test
+                    USING(model_id);''')
+        )
+
+    @property
+    def model_type(self):
+        return self.metadata['model_type']
+
+    @property
+    def hyperparameters(self):
+        return self.metadata['hyperparameters']
+
+    @property
+    def model_hash(self):
+        return self.metadata['model_hash']
+
+    @property
+    def train_matrix_uuid(self):
+        return self.metadata['train_matrix_uuid']
+
+    @property
+    def pred_matrix_uuid(self):
+        return self.metadata['matrix_uuid']
+
+    @property
+    def as_of_date(self):
+        return self.metadata['as_of_date']
+
+    def __repr__(self):
+        return (
+        str({'model_id': self.model_id,
+             'model_group': self.model_group_id,
+             'model_type': self.model_type,
+             'as_of_date': self.as_of_date.strftime("%B %d, %Y"),
+             'model_hyperparameters': self.hyperparameters})
+        )
+
+    @cachedproperty
+    def predictions(self):
+        preds = pd.read_sql(
+            f'''
+            SELECT model_id,
+                   entity_id,
+                   as_of_date,
+                   score,
+                   label_value,
+                   COALESCE(rank_abs, RANK() OVER(ORDER BY score DESC)) AS rank_abs,
+                   COALESCE(rank_pct, percent_rank()
+                   OVER(ORDER BY score DESC)) * 100 as rank_pct,
+                   test_label_timespan
+            FROM test_results.predictions
+            WHERE model_id = {self.model_id}
+            AND label_value IS NOT NULL
+            ''', con=self.engine)
+
+        return preds
+
+    @cachedproperty
+    def feature_importances(self, path):
```

Comment:
how is vanilla `cachedproperty` supposed to work with a `path` argument?

should it be cached at all? should these be two separate methods? or is `path` something which can optionally be turned into an object construction argument and object attribute?